File: app/src/main/java/com/pkfcooperparry/template/data/local/prefs/PreferencesHelper.java
package com.pkfcooperparry.template.data.local.prefs;


import com.pkfcooperparry.template.data.DataManager;

public interface PreferencesHelper {

    String getAccessToken();

    void setAccessToken(String accessToken);

    void setAccessTokenAndUserId(String accessToken, long userId);

    String getCurrentUserEmail();

    void setCurrentUserEmail(String email);

    Long getCurrentUserId();

    void setCurrentUserId(Long userId);

    String getCurrentUserName();

    void setCurrentUserName(String userName);

    int getCurrentUserLoggedInMode();

    void setCurrentUserLoggedInMode(DataManager.LoggedInMode mode);

    boolean isLoggedIn();

}

File: app/src/main/java/com/pkfcooperparry/template/data/local/prefs/PreferencesHelper.kt
package com.pkfcooperparry.template.data.local.prefs


import com.pkfcooperparry.template.data.DataManager

interface PreferencesHelper {

    var accessToken: String?

    var currentUserEmail: String?

    var currentUserId: Long?

    var currentUserName: String?

    val currentUserLoggedInMode: Int

    val isLoggedIn: Boolean

    fun setAccessTokenAndUserId(accessToken: String, userId: Long)

    fun setCurrentUserLoggedInMode(mode: DataManager.LoggedInMode)

}
